source/SIL.AppBuilder.Portal/common/public/workflow.ts

```diff
@@ -0,0 +1,254 @@
+import type {
+  AnyEventObject,
+  StateMachineDefinition,
+  StateMachine,
+  TransitionDefinition,
+  TransitionDefinitionMap,
+  StateNode as XStateNode
+} from 'xstate';
+import type { RoleId } from './prisma.js';
+
+export enum ActionType {
+  /** Automated Action */
+  Auto = 0,
+  /** User-initiated Action */
+  User
+}
+
+export enum AdminLevel {
+  /** NoAdmin/OwnerAdmin */
+  None = 0,
+  /** LowAdmin */
+  Low,
+  /** Approval required */
+  High
+}
```

Comment:
By 'switch the names around' do you mean just reorder them? Or changing the meaning? Because I'm pretty sure we agree on the meaning?

Reply:
I'm not sure how to make the name more descriptive without making all usage of it unnecessarily verbose. I can add more comments?

Reply:
I mean, is this enum used as a minimum required admin level for a user to complete an action, or as the level of admin that a user/action has, or something else? My understanding is the former, but the name implies to me the latter. The comments seem backwards to me. What does the comment "NoAdmin/OwnerAdmin" mean? That the action was performed by an admin? That the action should not require admin in any way, as indicated by the name "None"?

Does that make sense? I'd rather rename the enum to "RequiredAdminLevel" or something more descriptive like that. At least clarify the comments.

Reply:
The way I am using the enum, I am including a state or transition in a workflow *only* if the specified admin level of the workflow type (i.e. OwnerAdmin, LowAdmin, NoAdmin, Default, etc.) matches any of the specified admin level types in the meta (or if there is no specified level).

For Example:
A workflow will only include the `Readiness Check`, `Approval`, `Approval Pending`, and `Terminated States` if the workflow's admin level is `AdminLevel.High`, which is the case, for example, in the `SIL_Default_AppBuilders_Android_GooglePlay` scheme in DWKit. 

If a workflow's admin level is `AdminLevel.Low`, it will not include the states for the approval process, but there are some actions (i.e. transitions) that require an OrgAdmin.

If a workflow's admin level is `AdminLevel.None`, which is the case for any of the DWKit schemes marked as `OwnerAdmin` or `NoAdmin`, the transitions mentioned above that would require an OrgAdmin instead require the project owner.

Reply:
`RequiredAdminLevel` seems like a reasonable name. I will also clarify the comments with something along the lines of the above comment.

Reply:
Ok, I see. My suggestion might be `AdminVerificationLevel` then